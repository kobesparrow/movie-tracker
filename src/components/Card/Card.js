import React, {Component} from 'react';
import { Link } from 'react-router-dom'

export default class Card extends Component {
  
  render() {
    const imageSource = `https://image.tmdb.org/t/p/w500/${this.props.poster_path}`
    return (
      <Link to={`popular/${this.props.movie_id}`}>
          <img src={imageSource} alt='movie poster' className='movie-img' />
      </Link>
    )
  }
}






