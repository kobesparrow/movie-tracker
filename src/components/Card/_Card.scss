.movie-card {
  width: 30%;
  height: 300px;
  margin: 0px 5px;
  border-radius: 5px;
  transition-duration: .3s;
  border: 5px black;
}

.movie-img {
  height: 200px;
}