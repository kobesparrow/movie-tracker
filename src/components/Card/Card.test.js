import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme';

const mockMovie = {name: 'endgame', id: 123}

describe('Card', () => {
    let wrapper;


    beforeEach(() => {
        wrapper = shallow(
            <Card {...mockMovie} key={mockMovie.id} />
        )
    });

    it('should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    });
})