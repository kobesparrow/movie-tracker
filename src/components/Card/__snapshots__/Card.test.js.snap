// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
<Link
  to="popular/undefined"
>
  <img
    alt="movie poster"
    className="movie-img"
    src="https://image.tmdb.org/t/p/w500/undefined"
  />
</Link>
`;
